File: iosApp/iosApp/Services/NFTsServiceAssembler.swift
// Created by web3d4v on 16/05/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import web3walletcore

final class NFTsServiceAssembler: AssemblerComponent {
    
    func register(to registry: AssemblerRegistry) {
        
        registry.register(scope: .singleton) { resolver -> NFTsService in
            
            OpenSeaNFTsService(
                networksService: resolver.resolve(),
                store: KeyValueStore(name: "\(NFTsService.self)")
            )
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/NFTsServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3lib.keyValueStore.KeyValueStore
import com.sonsofcrypto.web3lib.services.networks.NetworksService
import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import com.sonsofcrypto.web3walletcore.services.nfts.NFTsService
import com.sonsofcrypto.web3walletcore.services.nfts.OpenSeaNFTsService
import smartadapter.internal.extension.name

class NFTsServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(NFTsService::class.name, AssemblerRegistryScope.SINGLETON) {
            OpenSeaNFTsService(
                it.resolve(NetworksService::class.name),
                it.resolve("${KeyValueStore::class.name}.NFTsService"),
            )
        }
    }
}